internal/overlord/servstate/handlers.go

```diff
@@ -83,6 +84,12 @@ func (m *ServiceManager) doStart(task *state.Task, tomb *tomb.Tomb) error {
 	cmd := exec.Command(args[0], args[1:]...)
 	cmd.SysProcAttr = &syscall.SysProcAttr{Setpgid: true}
 
+	// Pass service description's environment variables to child process
+	cmd.Env = os.Environ()
+	for _, v := range service.Environment {
+		cmd.Env = append(cmd.Env, v.Name+"="+v.Value)
+	}
```

Comment:
Thanks for fixing this, Ben.  We indeed need a good test for this sort of change.

Reply:
Ah, and it's also worth checking (and testing) what happens if there's a variable in the environment process which needs to be replaced by something explicit on the service.

Reply:
Added a test for this: it tests brand new env vars, env vars only set in the parent, and env vars in the parent but overridden in the layer config. I'll do any `$PATH` type of interpolation in a subsequent commit.